src/server/test/web/readingsBarMeterFlow.js

```diff
@@ -7,307 +7,417 @@
-                    // Load the expected response data from the corresponding csv file
-                    const expected = await parseExpectedCsv('src/server/test/web/readingsData/expected_bar_ri_15_mu_kW_gu_kW_st_-inf_et_inf_bd_1.csv');
-                    // Create a request to the API for unbounded reading times and save the response
-                    const res = await chai.request(app).get(`/api/unitReadings/bar/meters/${METER_ID}`)
-                        .query({
-                            timeInterval: ETERNITY.toString(),
-                            barWidthDays: 1,
-                            graphicUnitId: unitId
-                        });
-                    // Check that the API reading is equal to what it is expected to equal
-                    expectReadingToEqualExpected(res, expected);
-                });
+mocha.describe("readings API", () => {
+	mocha.describe(
+		"readings test, test if data returned by API is as expected",
+		() => {
+			mocha.describe("for bar charts", () => {
+				mocha.describe("for flow meters", () => {
+					mocha.it(
+						"B15: 13 day bars for 15 minute reading intervals and flow units with +-inf start/end time & kW as kW",
+						async () => {
+							const unitData = [
+								{
+									// u4
+									name: "kW",
+									identifier: "",
+									unitRepresent: Unit.unitRepresentType.FLOW,
+									secInRate: 3600,
+									typeOfUnit: Unit.unitType.UNIT,
+									suffix: "",
+									displayable: Unit.displayableType.ALL,
+									preferredDisplay: true,
+									note: "kilowatts",
+								},
+								{
+									// u5
+									name: "Electric",
+									identifier: "",
+									unitRepresent: Unit.unitRepresentType.FLOW,
+									secInRate: 3600,
+									typeOfUnit: Unit.unitType.METER,
+									suffix: "",
+									displayable: Unit.displayableType.NONE,
+									preferredDisplay: false,
+									note: "special unit",
+								},
+							];
+							const conversionData = [
+								{
+									// c4
+									sourceName: "Electric",
+									destinationName: "kW",
+									bidirectional: false,
+									slope: 1,
+									intercept: 0,
+									note: "Electric → kW",
+								},
+							];
+							const meterData = [
+								{
+									name: "Electric kW",
+									unit: "Electric",
+									defaultGraphicUnit: "kW",
+									displayable: true,
+									gps: undefined,
+									note: "special meter",
+									file: "test/web/readingsData/readings_ri_15_days_75.csv",
+									deleteFile: false,
+									readingFrequency: "15 minutes",
+									id: METER_ID,
+								},
+							];
+							// Load the data into the database
+							await prepareTest(unitData, conversionData, meterData);
+							// Get the unit ID since the DB could use any value.
+							const unitId = await getUnitId("kW");
+							// Load the expected response data from the corresponding csv file
+							const expected = await parseExpectedCsv(
```

Comment:
Nice if this was a single line per another comment. Note this applies to several changes in the file that makes it show up as changes.